packages/mui-base/src/ListboxUnstyled/defaultListboxReducer.test.ts

```diff
@@ -315,8 +315,13 @@ describe('useListbox defaultReducer', () => {
 
       const action: ListboxAction<string> = {
         type: ActionTypes.textNavigation,
-        isMatch: (option) => option === 'three',
-        startWithCurrentOption: false,
+        textCriteria: {
```

Comment:
Wouldn't it be sufficient if `textCriteria` was simply a string?
The logic would then be as follows:
- if the string contains just one character, move to the next option that starts with that character
- otherwise, check if the currently highlighted option matches the `textCriteria`. If it does, do nothing; if it does not, go to the next matching option.

Can you think of a case this would not work?